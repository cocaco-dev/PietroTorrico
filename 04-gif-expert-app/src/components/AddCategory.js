import React, {useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState(''); //Undefined si no se pone las comillas
    const handleInputChange = (e) => {        
        setInputValue(e.target.value)        
    }
    const handleSubmit = (e) => {
        e.preventDefault();   
        console.log('ENVIO', inputValue);     
        if(inputValue.trim().length > 2){
            setCategories(cate => [inputValue, ...cate]); //cate hace referencia al contenido del componente  
            setInputValue('');
        }
    }

    return (
        <form onSubmit = {handleSubmit}>        
            {/* <p>{inputValue}</p>     */}
            <input
                type="text"
                value = {inputValue}
                onChange={handleInputChange}
                placeholder = 'Ingresa una categoria'
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}