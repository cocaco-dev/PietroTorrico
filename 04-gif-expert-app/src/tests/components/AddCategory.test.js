import React from 'react';
import {shallow} from 'enzyme';
import { AddCategory } from '../../components/AddCategory';


describe('Pruebas en el componte <AddCategory/>', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories = {setCategories}/>); 
    beforeEach (() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories = {setCategories}/>);
    })

    test('probar con el snapshot ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target: {value} });
        const parrafo = wrapper.find('p').text().trim();
        expect(parrafo).toBe(value)

    })
    
    test('No debe de postear la información on submit ', () => {
        wrapper.find('form').simulate('submit', {preventDefault: ()=> {}});
        
        expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('debe de llamar el setCategories y limpiar la caja de texto ', () => {

        // 1.Simulació del InputChange
        const value = 'Hola Mundo';
        wrapper.find('input').simulate('change', {target: {value}});

        // 2.Simular el submit
        wrapper.find('form').simulate('submit', {preventDefault: ()=>{}})

        // 3. SetCategories se debe de haber llamado
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        // 4. El valor del inputo debe ser ''
        expect(wrapper.find('input').prop('value')).toBe('')
    })
    

    

})