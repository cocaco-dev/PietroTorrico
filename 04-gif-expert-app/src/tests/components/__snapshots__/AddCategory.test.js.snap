// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componte <AddCategory/> probar con el snapshot  1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
